src/main/java/com/smartsheet/api/internal/ReportResourcesImpl.java

```diff
@@ -0,0 +1,251 @@
+     *
+     * Exceptions:
+     *   InvalidRequestException : if there is any problem with the REST API request
+     *   AuthorizationException : if there is any problem with the REST API authorization(access token)
+     *   ResourceNotFoundException : if the resource can not be found
+     *   ServiceUnavailableException : if the REST API service is not available (possibly due to rate limiting)
+     *   SmartsheetRestException : if there is any other REST API related error occurred during the operation
+     *   SmartsheetException : if there is any other error occurred during the operation
+     *
+     * @param reportId the folder id
+     * @param includes the optional objects to include in response
+     * @param pageSize Number of rows per page
+     * @param page page number to return
+     * @return the report (note that if there is no such resource, this method will throw ResourceNotFoundException
+     * rather than returning null)
+     * @throws SmartsheetException the smartsheet exception
+     */
+    public Report getReport(long reportId, EnumSet<ReportInclusion> includes, Integer pageSize, Integer page) throws SmartsheetException{
+        this.checkParameters(pageSize, page);
+        String path = "reports/" + reportId;
+        HashMap<String, String> parameters = new HashMap<String, String>();
+        parameters.put("include", QueryUtil.generateCommaSeparatedList(includes));
+
+        if (pageSize != null) {
+            parameters.put("pageSize", pageSize.toString());
+        }
+
+        if (page != null) {
+            parameters.put("page", page.toString());
+        }
+
+        path += QueryUtil.generateUrl(null, parameters);
+        return this.getResource(path, Report.class);
+    }
+
+    /**
+     * Sends a report as a PDF attachment via email to the designated recipients.
+     *
+     * It mirrors to the following Smartsheet REST API method: POST /reports/{id}/emails
+     *
+     * Exceptions:
+     *   InvalidRequestException : if there is any problem with the REST API request
+     *   AuthorizationException : if there is any problem with the REST API authorization(access token)
+     *   ResourceNotFoundException : if the resource can not be found
+     *   ServiceUnavailableException : if the REST API service is not available (possibly due to rate limiting)
+     *   SmartsheetRestException : if there is any other REST API related error occurred during the operation
+     *   SmartsheetException : if there is any other error occurred during the operation
+     *
+     * @param reportId the report id
+     * @param email the recipient email
+     * @return the report (note that if there is no such resource, this method will throw ResourceNotFoundException
+     * rather than returning null)
+     * @throws SmartsheetException the smartsheet exception
+     */
+    public void sendSheet(long reportId, SheetEmail email) throws SmartsheetException{
```

Comment:
This should return a Result object: http://smartsheet-platform.github.io/api-docs/#send-report-via-email